### PR TITLE
Bulk issues

### PR DESCRIPTION
**Pull request by tejans24**
connects #3212 

Test plan in rails console on uat:

Try other test scenarios as seen appropriate:

```
# fetch an appeal with a specific id
appeal = Appeal.find_or_create_by_vacols_id("2479097")
appeal.issues
```

```
# get assignments of CF_SCOTTY_317
assignments = CaseAssignmentRepository.load_from_vacols("CF_SCOTTY_317", true)

# this should not make additional queries to vacols
assignments[0].issues
```

### CODE REVIEW

File: app/models/vacols/case_issue.rb [hunk 1]
```diff
@@ -13,62 +13,75 @@ class VACOLS::CaseIssue < VACOLS::Record
...
-          case when ISSREF.LEV2_CODE = '##' then
-            VFTYPES.FTDESC else ISSREF.LEV2_DESC
-          end
-        end ISSLEV2_LABEL,
-        case when ISSUES.ISSLEV3 is not null then
-          case when ISSREF.LEV3_CODE = '##' then
-            VFTYPES.FTDESC else ISSREF.LEV3_DESC
-          end
-        end ISSLEV3_LABEL
+    conn.transaction do
```
[the remark this answers is not in the document]
Comment (tejans24) [reply app/models/vacols/case_issue.rb]: Actually, we don't need it. Removed it.

File: app/models/vacols/case_issue.rb [hunk 2]
```diff
@@ -13,62 +13,75 @@ class VACOLS::CaseIssue < VACOLS::Record
...
+      SQL
+
+      issues_result = MetricsService.record("VACOLS: CaseIssue.issues for #{issue_keys}", name: "CaseIssue.issues",
+                                                                                          service: :vacols) do
+        conn.exec_query(sanitize_sql_array([query, issue_keys]))
+      end
+
+      issues_result.to_hash.reduce({}) do |memo, result|
+        issue_key = result["isskey"].to_s
+        memo[issue_key] ||= []
```
[the remark this answers is not in the document]
Comment (tejans24) [reply app/models/vacols/case_issue.rb]: Neat

File: app/services/case_assignment_repository.rb [hunk 3]
```diff
@@ -4,14 +4,18 @@ def self.load_from_vacols(css_id)
...
       active_cases_for_user = VACOLS::CaseAssignment.active_cases_for_user(css_id)
       active_cases_vacols_ids = active_cases_for_user.map(&:vacols_id)
       active_cases_aod_results = VACOLS::Case.aod(active_cases_vacols_ids)
+      active_cases_issues = VACOLS::CaseIssue.descriptions(active_cases_vacols_ids)
 
       active_cases_for_user.map do |assignment|
         appeal = Appeal.find_or_initialize_by(vacols_id: assignment.vacols_id)
         appeal.attributes = assignment.attributes
         appeal.aod = active_cases_aod_results[assignment.vacols_id]
+        appeal.issues = Issue.load_from_vacols(active_cases_issues[assignment.vacols_id])
```
[the remark this answers is not in the document]
Comment (tejans24) [reply app/services/case_assignment_repository.rb]: 😁😅

File: app/models/vacols/case_issue.rb [hunk 4]
```diff
@@ -67,8 +66,19 @@ def self.descriptions(issue_key)
...
+
+    issues_result = MetricsService.record("VACOLS: CaseIssue.issues for #{vacols_ids}", name: "CaseIssue.issues",
+                                                                                        service: :vacols) do
+      conn.exec_query(sanitize_sql_array([query, vacols_ids]))
+    end
+
+    issues_result.to_hash.reduce({}) do |memo, result|
+      issue_key = result["isskey"].to_s
+      memo[issue_key] = (memo[issue_key] || []) << result
+      memo
```
Comment (mdbenjam) [inline app/models/vacols/case_issue.rb]: Do we even need this line, I think the previous one should return the new array.

File: app/services/case_assignment_repository.rb [hunk 5]
```diff
@@ -7,11 +7,17 @@ def self.load_from_vacols(css_id)
...
+      active_cases_issues = VACOLS::CaseIssue.descriptions(active_cases_vacols_ids)
+      
+      active_cases_for_user.each_with_index.map do |assignment, index|
+        case_issues_hash_array = active_cases_issues[assignment.vacols_id]
 
-      active_cases_for_user.map do |assignment|
         appeal = Appeal.find_or_initialize_by(vacols_id: assignment.vacols_id)
         appeal.attributes = assignment.attributes
-        appeal.aod = active_cases_aod_results[assignment.vacols_id]
+        appeal.aod = active_cases_aod_results[index]
```
Comment (mdbenjam) [inline app/services/case_assignment_repository.rb]: This is back to what it was before... You should remove the index and use vacols_id

File: app/models/vacols/case_issue.rb (hunk 4 above)
Comment (tejans24) [reply app/models/vacols/case_issue.rb]: No, I generally like to put one for explicit return in ruby. But I can remove it.

Comment (tejans24) [reply app/models/vacols/case_issue.rb]: This case needs an explicit return

**Pull request merged**